internal/allocation/allocation_manager.go

```diff
@@ -83,7 +83,7 @@ func (m *Manager) CreateAllocation(fiveTuple *FiveTuple, turnSocket ipnet.Packet
 	a.RelayAddr = listener.LocalAddr()
 
 	a.lifetimeTimer = time.AfterFunc(lifetime, func() {
-		if err := listener.Close(); err != nil {
```

Comment:
Nice catch. Right now we don't ever clean up allocations (but we close the socket) Does this change make the tests now pass?

I think we leak the `listener` now though, we should probably put the cleanup in `(a *Allocation) Close()`

Reply:
I think `(a *Allocation) Close()` has include `listener` close logic, see the code [allocation.go#L186](https://github.com/pion/turn/blob/master/internal/allocation/allocation.go#L186). 

Change to `listener.Close()` , the tests pass yet because  the Allocation will be deleted in `(a *Allocation) packetHandler`, details to see [allocation.go#L217](https://github.com/pion/turn/blob/master/internal/allocation/allocation.go#L217)

Reply:
Should we leave this line unchanged then? Approving the PR but w/e you think is best go for it :) 

Maybe we can refactor the entire shut down/destroy logic!

Reply:
@Sean-Der  Just change `Allocation.lifetimeTimer` handle function to `m.DeleteAllocation()`, I think it's more clear, code review again please.